include/unifex/transform_done.hpp

```diff
@@ -0,0 +1,364 @@
+      });
+    startedOp_ = 0 + 1;
+  }
+
+  ~type() {
+    if (startedOp_ < 0) {
+      finalOp_.destruct();
+    } else if (startedOp_ > 0) {
+      sourceOp_.destruct();
+    }
+    startedOp_ = 0;
+  }
+
+  void start() & noexcept {
+    unifex::start(sourceOp_.get());
+  }
+
+private:
+  friend source_receiver;
+  friend final_receiver;
+
+  using source_op_t = operation_t<Source, source_receiver>;
+
+  using final_sender_t = std::invoke_result_t<Done&>;
+
+  using final_op_t = operation_t<final_sender_t, final_receiver>;
+
+  UNIFEX_NO_UNIQUE_ADDRESS Done done_;
+  UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
+  int startedOp_ = 0;
+  union {
+    manual_lifetime<source_op_t> sourceOp_;
+    manual_lifetime<final_op_t> finalOp_;
+  };
+};
+
+template<typename Source, typename Done>
+class _sndr<Source, Done>::type {
+  using final_sender_t = std::invoke_result_t<Done&>;
+
+public:
+  template<template<typename...> class Variant,
+           template<typename...> class Tuple>
+  using value_types = Variant<Tuple<>>;
+
+  template <template <typename...> class Variant>
+  using error_types = typename concat_type_lists_unique_t<
+      typename Source::template error_types<
+          decayed_tuple<type_list>::template apply>,
+      typename decltype(std::declval<Done>()())::template error_types<
+          decayed_tuple<type_list>::template apply>,
+      type_list<std::exception_ptr>>::template apply<Variant>;
+
+  template<typename Source2, typename Done2>
+  explicit type(Source2&& source, Done2&& done)
+    noexcept(
+      std::is_nothrow_constructible_v<Source, Source2> &&
+      std::is_nothrow_constructible_v<Done, Done2>)
+  : source_((Source2&&)source)
+  , done_((Done2&&)done)
+  {}
+
+  template<
+    typename Receiver,
+    typename SourceReceiver = receiver_type<Source, Done, Receiver>,
+    typename FinalReceiver = final_receiver_type<Source, Done, Receiver>,
+    std::enable_if_t<
+        std::is_move_constructible_v<Source> &&
+        std::is_move_constructible_v<Done> &&
+        std::is_move_constructible_v<Receiver> &&
+        is_connectable_v<Source&, SourceReceiver> &&
+        is_connectable_v<final_sender_t, FinalReceiver>, int> = 0>
+  operation_type<Source, Done, Receiver> connect(Receiver&& r) &&
+       noexcept(
+        std::is_nothrow_move_constructible_v<Source> &&
+        std::is_nothrow_move_constructible_v<Done> &&
+        std::is_nothrow_move_constructible_v<Receiver>) {
+    return operation_type<Source, Done, Receiver>{
+      (Source&&)source_, 
+      (Done&&)done_, 
+      (Receiver&&)r
+    };
+  }
+
+  template<
+    typename Receiver,
+    typename SourceReceiver = receiver_type<Source, Done, Receiver>,
+    typename FinalReceiver = final_receiver_type<Source, Done, Receiver>,
+    std::enable_if_t<
+        std::is_copy_constructible_v<Source> &&
+        std::is_copy_constructible_v<Done> &&
+        std::is_move_constructible_v<Receiver> &&
```

Comment:
This should be `std::is_constructible_v<std::remove_cvref_t<Receiver>, Receiver>`